pymmcore_plus/_client.py

```diff
@@ -39,10 +54,34 @@ def __init__(
 
         self._cb_thread = None
         self._callbacks = set()
-        if callback is not None:
-            self.register_callback(callback)
+        self.register_callback(_CBrelay(self))
 
     def register_callback(self, callback: CallbackProtocol):
+        """Register callback object in proxy process to receive remote callbacks.
+
+        Not to be confused with `mmcore.registerCallback`, which is only used once by
+        CMMCorePlus to receive events coming from the internal C++ CMMcore object.
+
+        Note: RemoteMMCore automatically behaves as a callback receiver, you can
+        connect to any of the signals in the `_CMMCoreSignaler` class:
+
+            proxy = RemoteMMCore()
+            proxy.systemConfigurationLoaded.connect(lambda: print("loaded!"))
+
+        This method can be used to register an independent callback object.
+
+        Parameters
+        ----------
+        callback : CallbackProtocol
```

Comment:
It's a bit confusing to use this in docstring as this isn't a public object that people can normally interact with or see documented. Maybe it should be imported in `__init__`?

Reply:
for exactly that reason, I tend not to worry too much about custom types in numpy docstrings.  at this point in the project having the class above  `class CallbackProtocol(Protocol):` feels like sufficient documentation to me
(I'd much prefer a world in which we all moved to using type annotations in signatures, and leaving them mostly out of docstrings).   exporting it in the main namespace is sort of an independent question (and if we're making this private, then i probably wouldn't want to do that). the one thing we could do (which is ugly, but accurate) would be to change it to `callback : pymmcore_plus._client.CallbackProtocol` ... but for reasons mentioned above, I'd probably skip that too.

Reply:
also, it says on the next line: "Just an object that has a `receive_core_callback` method"